site/content/docs/main/troubleshooting/common-errors.md

```diff
@@ -0,0 +1,65 @@
+# Troubleshooting common errors
+
+## Unexpected HTTP errors
+
+Here are some steps to take in investigating common HTTP errors that users may encounter.
+We'll include example error cases to debug with these steps.
+
+1. Inspect the HTTP response in detail (possibly via `curl -v`).
+
+    Here we're looking to validate if the error response is coming from the backend app, Envoy, or possibly another proxy in front of Envoy.
+    If the response has the `server: envoy` header set, the request at least made it to the Envoy proxy so we can likely rule out anything before it.
+    The error may originate from Envoy itself or the backend app.
+    Look for headers or a response body that may originate from the backend app to verify if the error is in fact just the intended app behavior.
+    In the example below, we can see the response looks like it originates from Envoy, based on the `server: envoy` header and response body string.
+
+    ```
+    curl -vvv example.projectcontour.io
+    ...
+    > GET / HTTP/1.1
+    > Host: example.projectcontour.io
+    ...
+    >
+    < HTTP/1.1 503 Service Unavailable
+    < content-length: 91
+    < content-type: text/plain
+    < vary: Accept-Encoding
+    < date: Tue, 06 Feb 2024 03:44:30 GMT
+    < server: envoy
+    <
+    * Connection #0 to host example.projectcontour.io left intact
+    upstream connect error or disconnect/reset before headers. reset reason: connection failure
+    ```
+
+1. Look at the Envoy pod logs for the access logs corresponding to the erroring request/response.
+
+    The exact fields/field ordering present in the access log may vary if you have [configured a custom access log string or JSON access logs][0].
+    For example for a Contour installation using the [default Envoy access log format][1] we would want to inspect:
+    * `%REQ(:METHOD)%`, `%REQ(X-ENVOY-ORIGINAL-PATH?:PATH)%`, `%REQ(:AUTHORITY)%`, `%PROTOCOL%`: Ensure these are sensible values based on your configured route and HTTP request
+    * `%RESPONSE_FLAGS%`: See the [documentation on Envoy response flags][2] and below how to interpret a few of them in a Contour context:
+      * `UF`: Likely that Envoy could not connect to the upstream
+      * `UH`: Upstream Service has no health/ready pods
+      * `NR`: No configured route matching the request
+    * `%DURATION%`: Can correlate this with any configured timeouts
+    * `%RESP(X-ENVOY-UPSTREAM-SERVICE-TIME)%`: Can correlate this with any configured timeouts. If `-` then this is a hint the request was never forwarded to the upstream.
+    * `%UPSTREAM_HOST%`: Can be used to verify the exact upstream instance/pod that might be erroring.
```

Comment:
we might want to mention here that is the pod ip more explicitly. I see it here but I think I get it because I know about it